text/0000-stable-rustdoc-urls.md

```diff
@@ -0,0 +1,259 @@
+
+- Rustdoc could add a `v.` prefix for items in the value namespace. This would be more consistent with the other namespaces, at the cost of making the URLs for functions slightly confusing (favoring criteria 2 over criteria 3).
+- Rustdoc could lengthen the prefixes to `type.` and `macro.`. This makes the URLs easier to read, at the cost of making them more confusing for traits (consider `type.Trait.html`).
+- Rustdoc could use the existing specific names only when there is no risk of a semver-compatible change being able to change the kind. This would need careful inspection to make sure there is in fact no risk. It would also be slightly inconsistent with other URLs.
+
+[#35236]: https://github.com/rust-lang/rust/pull/35236
+
+## Alternatives
+
+These alternatives are substantial changes to the RFC.
+
+- Rustdoc could stabilize the links it uses, but without keeping backwards compatibility by not generating `kind.name.html`. This has little benefit over the RFC,
+  other than slightly less disk space used and implementation complexity.
+- Rustdoc could keep the status quo. This can cause no naming conflicts on Windows, but has the drawback that links could silently break even for semver-compatible changes.
+- Rustdoc could choose to make URLs stable _neither_ across rustdoc versions nor the version of the code being documented,
+  for example by using `kind.name.SHA256SUM(rustdoc version).html`. This makes it more clear that the URLs are not intended to be stable,
+  at the cost of breaking links across much of the ecosystem.
+
+# Prior art
+[prior-art]: #prior-art
+
+- `go doc` generates all documentation on one page and uses URL hashes, without namespacing.
+   This causes conflicts when two items from different namespaces are in the same package.
+- `java` only allows classes at the top-level, so `javadoc` has no need for namespacing.
+   To distinguish between methods and fields, `javadoc` includes `()` in the URL fragment for methods.
+- `Racket` only allows functions at the top-level, and so has no need for namespacing.
+- `doxygen` names HTML pages after their C++ source files, and appends a random hash in the URL fragment to avoid namespace conflicts.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Is there a way to resolve the naming conflicts on Windows? If not, is that worth blocking the RFC, given there are existing conflicts?
+- Are there other semver-incompatibilities in the current pages that haven't been addressed?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+Rustdoc could stabilize page hashes:
+
+- Associated items for traits will use the same hash as for types, unless there is a conflict with the hash for a type.
+
+	A change from
+	```rust
+	struct S;
+	impl S { fn f() {} }
+	```
+	to
+	```rust
+	struct S;
+	trait T { fn f(); }
+	impl T for S { fn f() {} }
+	```
+	is semver compatible, but currently breaks the hash (it changes from `#method.f` to `#tymethod.f`).
```

Comment:
This seems definitely not semver compatible -- all users would need to now import T.

Reply:
Yes, you're right. I'll remove this whole section then, it can stay as `#method` and `#tymethod`.